### ISSUE TITLE
Code reloading makes page navigation impossible on Windows

### ISSUE THREAD

**Issue by lstor** [phoenixframework/phoenix#805]
#### Problem
When clicking a page link, live reload causes the current page to be re-rendered, effectively canceling navigation. Verified on a fresh project too.

#### Example
1. Let's say we're on a rendered `/users` view, and click to edit `/users/1`
2. The browser tries to navigate to `/users/1/edit`.
3. At the same time, the live reload triggers a call to `/users` (the page we're on). The page gets rendered and the attempted navigation is canceled.

#### Work-around
Changing the config to the following resolves the problem (at the cost of live reloading):
```elixir
config :app, App.Endpoint,
  live_reload: [
    patterns: [] # NOTE: Removed
  ]
```

Setting `watchers` to `[]` does *not* change the behavior.

#### Version info
Elixir 1.0.4
Erlang 17.5
Phoenix 0.11.0
Windows 7, 64-bit

**Comment by josevalim** [comment]
Someone reported this issue on the phoenix_live_reload project. I can send
a link tomorrow when I am not in my phone. :)

On Friday, April 24, 2015, Lars Storjord <notifications@github.com> wrote:

> Problem
>
> When clicking a page link, live reload causes the current page to be
> re-rendered, effectively canceling navigation. Verified on a fresh project
> too.
> Example
>
>    1. Let's say we're on a rendered /users view, and click to edit
>    /users/1
>    2. The browser tries to navigate to /users/1/edit.
>    3. At the same time, the live reload triggers a call to /users (the
>    page we're on). The page gets rendered and the attempted navigation is
>    canceled.
>
> Work-around
>
> Changing the config to the following resolves the problem (at the cost of
> live reloading):
>
> config :app, App.Endpoint,
>   live_reload: [
>     patterns: [] # NOTE: Removed
>   ]
>
> Setting watchers to [] does *not* change the behavior.
> Version info
>
> Elixir 1.0.4
> Erlang 17.5
> Phoenix 0.11.0
> Windows 7, 64-bit
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/805>.
>


-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Lead Developer

**Comment by lstor** [comment]
Do you mean phoenixframework/phoenix_live_reload#7? If so, the causes may be related, but the effects seem to be different. In my case, the page is only reloaded when a link is clicked.

In any case, I didn't realize phoenix_live_reload was its own project, it makes more sense to report the issue there. I'll close this one =)

**Comment by lstor** [comment]
Resolved by [phoenix_live_reload/5c8ffb2](http://github.com/phoenixframework/phoenix_live_reload/commit/5c8ffb2649bcd2c3ec6ab64e0e8204ac2f2414a7).
